src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs

```diff
@@ -0,0 +1,164 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Data.SqlClient;
+using System.Text.RegularExpressions;
+using Microsoft.Health.Fhir.Core.Features.Search.Expressions;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors.QueryGenerators
+{
+    internal abstract class SearchParameterQueryGenerator : DefaultExpressionVisitor<SqlQueryGenerator, SqlQueryGenerator>
+    {
+        private const string DefaultCaseInsensitiveCollation = "Latin1_General_100_CI_AI_SC";
+        private const string DefaultCaseSensitiveCollation = "Latin1_General_100_CS_AS";
+
+        private static readonly Regex LikeEscapingRegex = new Regex("[%!\\[\\]_]", RegexOptions.Compiled);
+
+        public override SqlQueryGenerator VisitSearchParameter(SearchParameterExpression expression, SqlQueryGenerator context)
+        {
+            short searchParamId = context.Model.GetSearchParamId(expression.Parameter.Url);
+            SmallIntColumn searchParamIdColumn = V1.SearchParam.SearchParamId;
+
+            context.StringBuilder
+                .Append(searchParamIdColumn)
+                .Append(" = ")
+                .AppendLine(context.Parameters.AddParameter(searchParamIdColumn, searchParamId).ParameterName)
+                .Append("AND ");
+
+            return expression.Expression.AcceptVisitor(this, context);
+        }
+
+        public override SqlQueryGenerator VisitMissingSearchParameter(MissingSearchParameterExpression expression, SqlQueryGenerator context)
+        {
+            short searchParamId = context.Model.GetSearchParamId(expression.Parameter.Url);
+            SmallIntColumn searchParamIdColumn = V1.SearchParam.SearchParamId;
+
+            context.StringBuilder
+                .Append(searchParamIdColumn)
+                .Append(" = ")
+                .AppendLine(context.Parameters.AddParameter(searchParamIdColumn, searchParamId).ParameterName);
+
+            return context;
+        }
+
+        public override SqlQueryGenerator VisitMultiary(MultiaryExpression expression, SqlQueryGenerator context)
+        {
+            if (expression.MultiaryOperation == MultiaryOperator.Or)
+            {
+                context.StringBuilder.Append('(');
```

Comment:
Nit: single quotes here but double quotes below (for the closing paren).  Make them consistent?  Same in SqlQueryGenerator.cs